subprojects/core/src/main/java/org/gradle/api/internal/collections/FilteredCollection.java

```diff
@@ -75,8 +75,8 @@ public boolean isEmpty() {
         if (collection.isEmpty()) {
             return true;
         } else {
-            for (T o : collection) {
-                if (accept(o)) {
+            for (Iterator<?> it = iterator(); it.hasNext();) {
```

Comment:
Can be replaced by `for(o : this) {}`

Reply:
Oh, good point. I will change this.

Reply:
Change done directly on master: https://github.com/gradle/gradle/commit/9f011c89468b96562c329c3a19dc44aa9935330a